src/sinol_make/commands/export/__init__.py

```diff
@@ -0,0 +1,287 @@
+                    shutil.copy(file_path, target_dir)
+
+        print('Generating tests...')
+        generated_tests = self.get_generated_tests()
+        print('Copying only handwritten tests...')
+        for ext in ['in', 'out']:
+            os.mkdir(os.path.join(target_dir, ext))
+            for test in glob.glob(os.path.join(os.getcwd(), ext, f'{self.task_id}*.{ext}')):
+                if package_util.extract_test_id(test) not in generated_tests:
+                    shutil.copy(test, os.path.join(target_dir, ext))
+
+    def create_files(self, target_dir: str, config: dict):
+        """
+        Creates required files in target directory (makefile.in).
+        :param target_dir: Directory to create files in.
+        :param config: Config dictionary.
+        """
+        with open(os.path.join(target_dir, 'makefile.in'), 'w') as f:
+            cxx_flags = '-std=c++17'
+            c_flags = '-std=c17'
+            if 'extra_compilation_args' in config:
+                if 'cpp' in config['extra_compilation_args']:
+                    cxx_flags += ' ' + ' '.join(config['extra_compilation_args']['cpp'])
+                if 'c' in config['extra_compilation_args']:
+                    c_flags += ' ' + ' '.join(config['extra_compilation_args']['c'])
+
+            f.write(f'MODE = wer\n'
+                    f'ID = {self.task_id}\n'
+                    f'SIG = sinolmake\n'
+                    f'\n'
+                    f'TIMELIMIT = {config["time_limit"]}\n'
+                    f'SLOW_TIMELIMIT = {4 * config["time_limit"]}\n'
+                    f'MEMLIMIT = {config["memory_limit"]}\n'
+                    f'\n'
+                    f'OI_TIME = oiejq\n'
+                    f'\n'
+                    f'CXXFLAGS += {cxx_flags}\n'
+                    f'CFLAGS += {c_flags}\n')
+
+    def compress(self, tmpdir, target_dir):
+        """
+        Compresses target directory to archive.
+        :param tmpdir: Temporary directory path.
+        :param target_dir: Target directory path.
+        :return: Path to archive.
+        """
+        archive = os.path.join(tmpdir, f'{self.task_id}.tgz')
+        with tarfile.open(archive, "w:gz") as tar:
+            tar.add(target_dir, arcname=os.path.basename(target_dir))
+        return archive
+
+    def copy_test(self, ocen_dir, filename):
+        """
+        Copies test to ocen directory.
+        :param ocen_dir: Ocen directory path.
+        :param filename: Test filename.
+        """
+        base = os.path.basename(filename)
+        ext = base.split('.')[-1]
+        out = os.path.join(ocen_dir, ext, base)
+        shutil.copyfile(filename, out)
+
+    def ocen_gen_conf(self, ocen_dir, nums):
+        """
+        Generates ocen configuration file.
+        :param ocen_dir: Ocen directory path.
+        :param nums: Set of test numbers.
+        """
+
+        print("Generating ocen configuration file...")
+        with open(os.path.join(ocen_dir, 'oi.conf'), "w") as conf:
```

Comment:
Oh wow, this file exists and sio2 uses it? I didn't know it.

Reply:
No, this files is used by the “ocen” script on the contestants’ computers. It tells the script what tasks and tests there are. The constants are also hard cored in our original scripts for creating ocen packages

Reply:
The constants can be changed by the contestants